bvm/ballerina-core/src/main/java/org/ballerinalang/channels/DatabaseUtils.java

```
@@ -0,0 +1,324 @@
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BByte;
+import org.ballerinalang.model.values.BFloat;
+import org.ballerinalang.model.values.BInteger;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Locale;
+
+/**
+ * Utility methods for storing/fetching channel messages.
+ *
+ * @since 0.982.0
+ */
+public class DatabaseUtils {
+
+    private static HikariDataSource hikariDataSource;
+    public static HikariConfig config;
+    private static ConfigRegistry registry = ConfigRegistry.getInstance();
+    private static final String H2_MEM_URL = "jdbc:h2:mem:" + ChannelConstants.DB_NAME;
+
+    public static void addEntry(String channelName, BValue key, BValue value, BType keyType, BType valType) {
+
+        Connection con = null;
+        PreparedStatement stmt = null;
+        try {
+            con = getDBConnection();
+            stmt = con.prepareStatement(ChannelConstants.INSERT);
+            stmt.setString(1, channelName);
+            if (keyType != null) {
+                setParam(stmt, key, keyType, 2);
+            } else {
+                stmt.setNull(2, Types.VARCHAR);
+            }
+
+            setParam(stmt, value, valType, 3);
+            stmt.execute();
+        } catch (SQLException e) {
+            throw new BallerinaException("error in saving received channel message " + e.getMessage(),
+                    e);
+        } finally {
+            try {
+                if (stmt != null) {
+                    stmt.close();
+                }
+
+                if (con != null) {
+                    con.close();
+                }
+            } catch (SQLException e) {
+                //ignore
```

Shall we throw an error here?

---

The Exception is when trying to close a connection. We do not need to stop the execution in such a situation right?

---

Yes. But can this lead to exhaustion in connection pool?